test/suite/wtbackup.py

```diff
@@ -319,10 +326,36 @@ def take_incr_backup_block(self, bkup_c, newfile, backup_incr_dir, consolidate):
                 self.range_copy(newfile, offset, size, backup_incr_dir, consolidate)
                 lens.append(size)
                 did_work = True
+                # Check backup stats.
+                blocks2 = self.backup_get_stat(stat.conn.backup_blocks)
+                blocks_diff = blocks2 - blocks1
+                self.pr(f"{newfile}: Start {blocks1} range copy end {blocks2}, total blocks {blocks_diff}")
+
+                # Connection stats should track for compressed and uncompressed should
+                # track the total changed.
+                comp = self.backup_get_stat(stat.conn.backup_blocks_compressed)
+                uncomp = self.backup_get_stat(stat.conn.backup_blocks_uncompressed)
+                self.assertEqual(blocks2, comp + uncomp)
+                # Now check the same data source stats. The change for this range copy
+                # should track the difference for this file.
+                comp = self.backup_get_stat(stat.dsrc.backup_blocks_compressed, newfile)
+                uncomp = self.backup_get_stat(stat.dsrc.backup_blocks_uncompressed, newfile)
+                comp_diff = comp - last_comp
+                uncomp_diff = uncomp - last_uncomp
+                if blocks_diff == 0:
+                    self.assertEqual(0, comp_diff + uncomp_diff)
+                else:
+                    self.assertNotEqual(0, comp_diff + uncomp_diff)
```

Comment:
Could this become
```suggestion
                self.assertEqual(blocks_diff, comp_diff + uncomp_diff)
```
?
It looks like we perform the same check a few lines below, but only when `did_work` is true

Reply:
I originally had that and although all the backup tests passed, the compact tests that do backup were failing. I'll investigate more what compact is doing that causes the failure.

Reply:
Fixed